neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -558,6 +604,7 @@ func (vm *VirtualMachine) Cleanup() {
 	vm.Status.Node = ""
 	vm.Status.CPUs = nil
 	vm.Status.MemorySize = nil
+	vm.Status.MemoryProvider = nil
```

Comment:
There's a couple reasons I went for this approach. This was my thought process, lmk what you think:

1. We need _some_ way to know how to scale the memory on the runner pod. This can either be a field in the spec or status.
2. We don't want to _need_ the control plane to set it; that way, we get control over rollout. So, as you mentioned -- either spec or status, but we have to auto-set _something_.
3. IMO: We should have an escape hatch, once virtio-mem is deployed. I.e., for a particular VM, we should be able to say "hey, please use DIMM slots instead". Eventually this should be a project/endpoint/compute setting (until we decide it's stable enough), and then we can remove all the old code. This belongs in the spec.
4. IMO: Once the VM is up and running, we _shouldn't_ allow changes to MemoryProvider setting. Actually, it needs to be _before_ the VM is up and running, to prevent race conditions. This can either be done with a status field (because special permissions are required), or with webhook update validation (i.e. only allow updates if pending and the podName wasn't set). We've had [issues around rollout of new fields + webhook validation in the past](https://www.notion.so/neondatabase/d66861b2d8a8444887282d8558428791?pvs=4#3ddf05468a8a44378b4c9931ad3dca95), so I'm hesitant to go that route.
5. IMO: _it would be nice_ if forcing VMs to pick up a change in MemoryProvider could just be done by deleting the pods. Otherwise we have to do object "delete + recreate with modified spec" (which is kind of a pain). I'm thinking of e.g. "oh no, there's an issue with virtio-mem, we need to switch out all affected runner pods". This requires a separate field in the status, if we also want to respect the spec.
6. IMO: _it would be nice_ if we could avoid neonvm-controller updating the spec (because I view "spec" as kind of "here's what the user asked", so to me it feels a little odd to update the spec. but... we have defaults anyways, so maybe this is analogous to that?)

From going through it again, I think it would be _fine_ to just use the spec, but I do also like having the status be separate. With (5) I'm probably just being paranoid.

WDYT?